jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -0,0 +1,628 @@
+//
+
+package org.eclipse.jetty.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.Flow;
+import java.util.function.BiPredicate;
+
+import org.eclipse.jetty.io.content.ContentSinkOutputStream;
+import org.eclipse.jetty.io.content.ContentSinkSubscriber;
+import org.eclipse.jetty.io.content.ContentSourceInputStream;
+import org.eclipse.jetty.io.content.ContentSourcePublisher;
+import org.eclipse.jetty.io.content.ContentSourceTransformer;
+import org.eclipse.jetty.io.internal.ByteBufferChunk;
+import org.eclipse.jetty.io.internal.ContentCopier;
+import org.eclipse.jetty.io.internal.ContentSourceByteBuffer;
+import org.eclipse.jetty.io.internal.ContentSourceConsumer;
+import org.eclipse.jetty.io.internal.ContentSourceString;
+import org.eclipse.jetty.util.Blocking;
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.FutureCallback;
+import org.eclipse.jetty.util.FuturePromise;
+import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.Promise;
+
+/**
+ * <p>Namespace class that contains the definitions of a {@link Source content source},
+ * a {@link Sink content sink} and a {@link Chunk content chunk}.</p>
+ */
+public class Content
+{
+    private Content()
+    {
+    }
+
+    /**
+     * <p>Copies the given content source to the given content sink, notifying
+     * the given callback when the copy is complete.</p>
+     *
+     * @param source the source to copy from
+     * @param sink the sink to copy to
+     * @param callback the callback to notify when the copy is complete
+     */
+    public static void copy(Source source, Sink sink, Callback callback)
+    {
+        copy(source, sink, callback, null);
+    }
+
+    /**
+     * <p>Copies the given content source to the given content sink, notifying
+     * the given callback when the copy is complete.</p>
+     * <p>The optional {@code chunkHandler} parameter is a predicate whose code
+     * may inspect the chunk and handle it differently from how the implementation
+     * would handle it.</p>
+     * <p>If the predicate returns {@code true}, it means that the chunk is handled
+     * externally and its callback completed, or eventually completed.</p>
+     * <p>If the predicate returns {@code false}, it means that the chunk is not
+     * handled, its callback will not be completed, and the implementation will
+     * handle the chunk and its callback.</p>
+     *
+     * @param source the source to copy from
+     * @param sink the sink to copy to
+     * @param callback the callback to notify when the copy is complete
+     * @param chunkHandler an optional predicate to handle the current chunk and its callback
+     */
+    public static void copy(Source source, Sink sink, Callback callback, BiPredicate<Chunk, Callback> chunkHandler)
```

Comment:
Maybe have a version like:
```suggestion
    public static void copy(Source source, Sink sink, Callback callback)
```